src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -83,6 +83,8 @@
     private boolean allow_merge_commit;
     private boolean allow_rebase_merge;
 
+    private boolean delete_branch_on_merge;
```

Comment:
Wait, does camel case not work?  It should.

Reply:
I prefer camelCase but, GHRepository require to fields to have same name as [Repository in Github](https://developer.github.com/v3/repos/)
So, I used `delete_branch_on_merge`